server/src/modules/auth/strategies/refresh-token.strategy.ts

```diff
@@ -0,0 +1,27 @@
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Request } from 'express';
+import { Injectable } from '@nestjs/common';
+import { AppConfigService } from 'src/config/app-config.service';
+import { JwtPayload } from '../types';
+
+@Injectable()
+export class RefreshTokenStrategy extends PassportStrategy(
+  Strategy,
+  'jwt-refresh',
+) {
+  constructor(private appConfigService: AppConfigService) {
+    super({
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+      secretOrKey: appConfigService.get<string>('JWT_REFRESH_SECRET'),
+      passReqToCallback: true,
+    });
+  }
+
+  validate(req: Request, payload: JwtPayload) {
```

Comment:
I suggetst validating payload to invalidate refresh token if user or refresh token fields was changed